app/models/reservation.rb

```diff
@@ -6,4 +6,19 @@ class Reservation < ApplicationRecord
   has_many :tickets, dependent: :destroy
 
   enum :status, %i[canceled booked confirmed]
+
+  scope :relevant_for, lambda { |user|
+    joins(:user, :show)
+    .where(user:)
+    .where(show: { start_time: Time.zone.now.. })
+    .where.not(status: :canceled)
+    .order(:start_time)
+  }
+
+  scope :for, lambda { |show|
```

Comment:
hmm this scope name doesn't tell too much

Reply:
Ok, I understand that naming things like that can be a little bit risky, however in context it reads quite well:

```ruby
show = Show.find(some_id)
reservations = Reservation.for(show)
```

Reply:
but in this query you are doing more than just searching for reservations assigned to specific show. Just `for` doesn't even tell with which table you are joining. It would be more readable if you split it into:
```ruby
scope :for_show, lambda {|show| joins(:show).where(show:) }
scope :not_canceled, -> { .where.not(status: :canceled).order(:status) }

Reservation.for_show(show).not_canceled
```

Thanks to that you have two reusable scopes and they tell way more about what's going on. Remember one important thing -  you do not write code for yourself but for others. Try to look at your code and especially names / variables as it tells what's under the hood and another person doesn't have to look for the definition :)